src/core/Akka.Cluster.Tests/MemberOrderingModelBasedTests.cs

```diff
@@ -0,0 +1,250 @@
+        {
+            if (model.AllMembers.Count == 0)
+                return AddNewMember.Generator();
+            return Gen.OneOf(ChangeMemberStatus.Generator(model.AllMembers.Keys), AddNewMember.Generator());
+        }
+
+        public override Arbitrary<Setup<MembershipState, MembershipModel>> Setup => Arb.From(Arb.Generate<MembershipSetup>()
+            .Select(x => (Setup<MembershipState, MembershipModel>)x)); // compiler ceremony :(
+
+        #region Setup
+
+        public class MembershipSetup : Setup<MembershipState, MembershipModel>
+        {
+            private readonly Member[] _members;
+
+            public MembershipSetup(UniqueAddress[] addresses)
+            {
+                // filter out any duplicates
+                _members =
+                    addresses.Distinct()
+                        .Select(x => new Member(x, int.MaxValue, MemberStatus.Up, ImmutableHashSet<string>.Empty))
+                        .ToArray();
+            }
+
+            public override MembershipState Actual()
+            {
+                return new MembershipState() { Members = ImmutableSortedSet<Member>.Empty.Union(_members) };
+            }
+
+            public override MembershipModel Model()
+            {
+                return new MembershipModel(_members.ToImmutableDictionary(x => x.Address, x => x));
+            }
+
+            public override string ToString()
+            {
+                return $"{GetType()}(Members = [{string.Join(",", _members.Select(x => x.ToString()))}]";
+            }
+        }
+
+        #endregion
+
+        #region Operations
+
+        public class ChangeMemberStatus : Operation<MembershipState, MembershipModel>
+        {
+            public static Gen<Operation<MembershipState, MembershipModel>> Generator(IEnumerable<Address> addresses)
+            {
+                var statusGen = ClusterGenerators.MemberStatusGenerator().Generator;
+                Func<Address, MemberStatus, Operation<MembershipState, MembershipModel>> generator =
+                    (address, status) => new ChangeMemberStatus(address, status);
+
+                var producer = FsharpDelegateHelper.Create(generator);
+
+                return Gen.Map2(producer, Gen.Elements(addresses), statusGen);
+            }
+
+            public readonly Address TargetedMember;
+            public readonly MemberStatus NewStatus;
+
+            public ChangeMemberStatus(Address targetedMember, MemberStatus newStatus)
+            {
+                TargetedMember = targetedMember;
+                NewStatus = newStatus;
+            }
+
+            public override bool Pre(MembershipModel model)
+            {
+                var containsMember = model.AllMembers.ContainsKey(TargetedMember);
+                if (!containsMember) return false; // short-circuit
+
+                var m = model.AllMembers[TargetedMember];
+                return Member.AllowedTransitions[m.Status].Contains(NewStatus);
```

Comment:
Verifies that the randomly generated transition is valid. I.e. a node can't go from `Joining --> Up`.